### PR TITLE
[ci]: Support the rpc image for mellanox

### PR DESCRIPTION
**Pull request by xumia**
#### Why I did it
Support the rpc image for mellanox.

Jenkins Pipeline reference:
https://github.com/Azure/sonic-build-tools/blob/master/jenkins/mellanox/buildimage-mlnx-202012-rpc/Jenkinsfile
https://github.com/Azure/sonic-build-tools/blob/master/jenkins/mellanox/buildimage-mlnx-201911-rpc/Jenkinsfile

#### How I did it
Rename the sync_rpc_image to docker_syncd_rpc_image
Add a new parameter syncd_rpc_image to build sonic-mellanox-rpc.bin

#### How to verify it
See the artifacts in https://dev.azure.com/mssonic/build/_build/results?buildId=20099&view=results

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [x] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

File: .azure-pipelines/azure-pipelines-build.yml [hunk 1]
```diff
@@ -115,6 +115,10 @@ jobs:
             fi
             if [ $(sync_rpc_image) == yes ]; then
               make $BUILD_OPTIONS ENABLE_SYNCD_RPC=y target/docker-syncd-$(platform_rpc)-rpc.gz
+              if [ $PLATFORM_AZP == mellanox ]; then
```
Comment (lguohan) [inline .azure-pipelines/azure-pipelines-build.yml:118]: i do not quite understand why we are injecting mellanox specific logic in here.

Comment (xumia) [reply .azure-pipelines/azure-pipelines-build.yml:118]: @lguohan , in Jenkins, we have a separate job with RPC enabled for mellanox only. Prefer to use the same job for the same platform. For one platform, we can have multiple images by design. Mellanox still needs the image with rpc enabled and the one with rpc none-enabled.

Comment (xumia) [reply .azure-pipelines/azure-pipelines-build.yml:118]: See https://sonic-jenkins.westus2.cloudapp.azure.com/job/mellanox/job/buildimage-mlnx-202012-rpc/ as an example.

Comment (lguohan) [reply .azure-pipelines/azure-pipelines-build.yml:118]: then introduce a new variable to build rpc image

Comment (xumia) [reply .azure-pipelines/azure-pipelines-build.yml]: sync_rpc_image is a little misleading, rename to syncd_rpc_image. Use the new parameter syncd_rpc_image to build the bin image.

**Review by lguohan — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by abdosi** [comment]
@xumia have conflict. create pr for 201911.

**Pull request merged**